staging/src/k8s.io/kubectl/pkg/cmd/debug/profiles/options.go

```diff
@@ -0,0 +1,193 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package profiles
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type DebugPodOptions struct {
+	// Container level
+	ContainerName   string
+	Args            []string
+	Image           string
+	ImagePullPolicy corev1.PullPolicy
+	Env             []corev1.EnvVar
+	TTY             bool
+	Stdin           bool
+	SetImages       map[string]string
+	ArgsOnly        bool
+
+	// Pod level
+	NodeName              string
+	PodName               string
+	Namespace             string
+	Annotations           map[string]string
+	ShareProcessNamespace *bool
+
+	SourcePod *corev1.Pod
+}
+
+func (o *DebugPodOptions) Build() (*corev1.Pod, error) {
```

Comment:
I know it's a draft, but it'd help me review if the https://github.com/golang/go/wiki/CodeReviewComments#doc-comments were already in place.

I think this is the boilerplate for creating a pod that will be common between different profiles. That SGTM and we can keep it even if we get rid of the DebugTypeOptions concept.

Reply:
I am really sorry for confusing you.. I have added some comments, could you take another look?

> I think this is the boilerplate for creating a pod that will be common between different profiles

Exactly!

Reply:
There's so much boilerplate it makes me think the interface isn't quite right. Thinking about it a bit more, what's interesting about profiles is less about what's common between them and more about the difference between them. Maybe profiles should mutate rather than generate. What do you think about something like https://github.com/verb/kubernetes/commit/9ee09c6e43d10b78eb3dc1b05b461751cee4921e ?

Reply:
@verb Hi! Your alternative looks great, which reduce the code massively. I have updated the code as you said, could you please take another look?